src/Microsoft.DotNet.ProjectModel/ProjectContext.cs

```diff
@@ -146,15 +167,12 @@ public ProjectContext CreateRuntimeContext(IEnumerable<string> runtimeIdentifier
                 return this;
             }
 
-            var standalone = !RootProject.Dependencies
-                .Any(d => d.Type.Equals(LibraryDependencyType.Platform));
-
             var context = CreateBuilder(ProjectFile.ProjectFilePath, TargetFramework)
-                .WithRuntimeIdentifiers(standalone ? runtimeIdentifiers : Enumerable.Empty<string>())
+                .WithRuntimeIdentifiers(runtimeIdentifiers)
                 .WithLockFile(LockFile)
                 .Build();
 
-            if (standalone && context.RuntimeIdentifier == null)
+            if (!context.IsPortable && context.RuntimeIdentifier == null)
```

Comment:
if `!context.IsPortable == context.IsStandalone` I would agree.

It's my understanding that this isn't the case though, and anywhere we want to do a standalone check we also need to check the runtime identifiers. This seems like we will disperse this knowledge throughout the code instead of centralizing it.